include/drivers/espi_emul.h

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_ESPI_SPI_EMUL_H_
+#define ZEPHYR_INCLUDE_DRIVERS_ESPI_SPI_EMUL_H_
+
+/**
+ * @file
+ *
+ * @brief Public APIs for the eSPI emulation drivers.
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/**
+ * @brief eSPI Emulation Interface
+ * @defgroup espi_emul_interface eSPI Emulation Interface
+ * @ingroup io_emulators
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define EMUL_ESPI_HOST_CHIPSEL 0
+
+struct espi_emul;
+
+/**
+ * Passes eSPI virtual wires set request to the emulator.
+ * The emulator updates the level of vw.
+ *
+ * @param emul Emulator instance
+ * @param vw The signal to be set.
+ * @param level The level of the signal to be set.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO General input / output error.
+ */
+typedef int (*emul_espi_api_set_vw)(struct espi_emul *emul,
+				    enum espi_vwire_signal vw,
+				    uint8_t level);
+
+/**
+ * Passes eSPI virtual wires get request to the emulator.
+ * The emulator returns the level of vw.
+ *
+ * @param emul Emulator instance
+ * @param vw The signal to be get.
+ * @param level The level of the signal to be get.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO General input / output error.
+ */
+typedef int (*emul_espi_api_get_vw)(struct espi_emul *emul,
+				    enum espi_vwire_signal vw,
+				    uint8_t *level);
+
+/**
+ * Find an emulator present on a eSPI bus
+ *
+ * At present the function is used only to find an emulator of the host
+ * device. It may be useful in systems with the SPI flash chips.
```

Comment:
Does this mean SPI or eSPI?

What does chip-select mean in the context of eSPI?

Reply:
SPI flash, it can be used with the eSPI bus.
Chip-select means the same as for SPI. Naturally, it is not needed for the single master - single slave system.
Please [see](https://www.intel.com/content/dam/support/us/en/documents/software/chipset-software/327432-004_espi_base_specification_rev1.0_cb.pdf) figure 2 and 6.